PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = 'digest.digest'
-    _inherit = "res.company"
-    _inherit = 'res.config.settings'
-    _inherit = 'sale.order'
-    _inherit = "sale.order.line"

Total Changes: 18

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/sale_management/models/digest.py
+++ b/addons/sale_management/models/digest.py
@@ -5,8 +5,8 @@ from odoo import fields, models, _
 from odoo.exceptions import AccessError
 
 
-class Digest(models.Model):
-    _inherit = 'digest.digest'
+class DigestDigest(models.Model):
+    _inherit = ['digest.digest']
 
     kpi_all_sale_total = fields.Boolean('All Sales')
     kpi_all_sale_total_value = fields.Monetary(compute='_compute_kpi_sale_total_value')
@@ -24,6 +24,6 @@ class Digest(models.Model):
         )
 
     def _compute_kpis_actions(self, company, user):
-        res = super(Digest, self)._compute_kpis_actions(company, user)
+        res = super()._compute_kpis_actions(company, user)
         res['kpi_all_sale_total'] = 'sale.report_all_channels_sales_action&menu_id=%s' % self.env.ref('sale.sale_menu_root').id
         return res

--- a/addons/sale_management/models/res_company.py
+++ b/addons/sale_management/models/res_company.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = "res.company"
+    _inherit = ["res.company"]
     _check_company_auto = True
 
     sale_order_template_id = fields.Many2one(

--- a/addons/sale_management/models/res_config_settings.py
+++ b/addons/sale_management/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = 'res.config.settings'
+    _inherit = ['res.config.settings']
 
     group_sale_order_template = fields.Boolean(
         "Quotation Templates", implied_group='sale_management.group_sale_order_template')

--- a/addons/sale_management/models/sale_order.py
+++ b/addons/sale_management/models/sale_order.py
@@ -9,7 +9,7 @@ from odoo.tools import is_html_empty
 
 
 class SaleOrder(models.Model):
-    _inherit = 'sale.order'
+    _inherit = ['sale.order']
 
     sale_order_template_id = fields.Many2one(
         comodel_name='sale.order.template',

--- a/addons/sale_management/models/sale_order_line.py
+++ b/addons/sale_management/models/sale_order_line.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class SaleOrderLine(models.Model):
-    _inherit = "sale.order.line"
+    _inherit = ["sale.order.line"]
     _description = "Sales Order Line"
 
     sale_order_option_ids = fields.One2many('sale.order.option', 'line_id', 'Optional Products Lines')

--- a/addons/sale_management/models/sale_order_option.py
+++ b/addons/sale_management/models/sale_order_option.py
@@ -5,7 +5,6 @@ from odoo.exceptions import UserError
 
 
 class SaleOrderOption(models.Model):
-    _name = 'sale.order.option'
     _description = "Sale Options"
     _order = 'sequence, id'
 

--- a/addons/sale_management/models/sale_order_template.py
+++ b/addons/sale_management/models/sale_order_template.py
@@ -5,7 +5,6 @@ from odoo.exceptions import ValidationError
 
 
 class SaleOrderTemplate(models.Model):
-    _name = "sale.order.template"
     _description = "Quotation Template"
     _order = 'sequence, id'
 

--- a/addons/sale_management/models/sale_order_template_line.py
+++ b/addons/sale_management/models/sale_order_template_line.py
@@ -5,7 +5,6 @@ from odoo.exceptions import UserError
 
 
 class SaleOrderTemplateLine(models.Model):
-    _name = "sale.order.template.line"
     _description = "Quotation Template Line"
     _order = 'sale_order_template_id, sequence, id'
 

--- a/addons/sale_management/models/sale_order_template_option.py
+++ b/addons/sale_management/models/sale_order_template_option.py
@@ -5,7 +5,6 @@ from odoo import api, fields, models
 
 
 class SaleOrderTemplateOption(models.Model):
-    _name = "sale.order.template.option"
     _description = "Quotation Template Option"
     _check_company_auto = True
 
